awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_python(cls, py_provider):
```

Comment:
I don't love this name. Maybe "new_custom" or something? I dunno. @graebm is the naming hero around her.